src/interfaces/chain.cpp

```diff
@@ -301,10 +296,10 @@ class ChainImpl : public Chain
         auto it = ::mempool.GetIter(txid);
         return it && (*it)->GetCountWithDescendants() > 1;
     }
-    void relayTransaction(const uint256& txid) override
+    bool broadcastTransaction(const CTransactionRef& tx, uint256& txid, std::string& err_string, const CAmount& max_tx_fee, bool relay) override
     {
-        CInv inv(MSG_TX, txid);
-        g_connman->ForEachNode([&inv](CNode* node) { node->PushInventory(inv); });
+        const TransactionError err = BroadcastTransaction(tx, txid, err_string, max_tx_fee, relay, false);
+        return TransactionError::OK == err || TransactionError::P2P_DISABLED == err;
```

Comment:
This is exactly right, but I think it could do with a comment. Something like:

```
// Chain clients only care about failures to accept the tx to the mempool. Disregard non-mempool related failures
// Note: this will need to be updated if BroadcastTransaction() is updated to return other non-mempool failures
// that Chain clients do not need to know about.
```

Reply:
It makes sense to filter some error messages, but for those that do make it through, at least the `err_string` - but probably also `TransactionError` - is useful for the client. So maybe it's better to change `TransactionError::OK` and erase the error message for errors that we don't want to propagate.